VIM highlighting

ixzzd opened the pull request:
In this PR I am working on highlighting for vim.
The idea is to highlight `%cx` and `%styled` ppx payload with `syntax/css.vim` syntax rules

davesnx commented:
Hey @ixzzd, Thanks for working on this!

Regarding your doubts on "Better interpolation". Interpolation is between `$(` and `)` and currently are the same as 
OCaml idents plus DOT (so we can have `Module.name`, `name`, etc), in any case should look like: `{ (A…Z) ∣ (a…z) ∣ 0…9 ∣ _ ∣ . }` (eventually will support `'` as well).

davesnx commented on packages/editors/vim/after/syntax/ocaml/styled.vim:
```diff
@@ -0,0 +1,20 @@
+
+syntax region styledPpxEncl matchgroup=ocamlPpxEncl start=+\[%cx+ end=+\]+ keepend contains=styledBody
+syntax region styledPpxEncl matchgroup=ocamlPpxEncl start=+\[%styled+ end=+\]+ keepend contains=styledBody
+syntax region styledBody start=+{|+ end=+|}+ contains=@CssSyntax contained
```
Start can either be `{|` or `"`. It's a regular OCaml string, which doesn't matter what's the termination.

ixzzd replied:
Updated in the https://github.com/ahrefs/vim-styled-ppx

davesnx commented on packages/editors/vim/after/syntax/ocaml/styled.vim:
```diff
@@ -0,0 +1,20 @@
+endif
+
+let b:styled_nested_syntax = 1
+syn include @CssSyntax syntax/css.vim
```
Is there any way we can extend syntax/css.vim and add the interpolation? 
If it's not, does it sound a good idea to fork it? :S

ixzzd replied:
Yes, I think it does make sense to include interpolation to the CSS highlighting rules.
Added here: https://github.com/ahrefs/vim-styled-ppx/blob/main/syntax/styled-ppx-payload.vim#L524-L525

ixzzd commented:
@davesnx I updated the PR- moved vim plugin to the separate repo to support vim plugin managers and added https://github.com/ahrefs/vim-styled-ppx link to the documentation
